warehouse/templates/email/token-compromised-leak/body.html

```diff
@@ -0,0 +1,68 @@
+{#
+ # Licensed under the Apache License, Version 2.0 (the "License");
+ # you may not use this file except in compliance with the License.
+ # You may obtain a copy of the License at
+ #
+ # http://www.apache.org/licenses/LICENSE-2.0
+ #
+ # Unless required by applicable law or agreed to in writing, software
+ # distributed under the License is distributed on an "AS IS" BASIS,
+ # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ # See the License for the specific language governing permissions and
+ # limitations under the License.
+-#}
+{% extends "email/_base/body.html" %}
+
+{% set site_name=request.registry.settings["site.name"] %}
+
+{% block content %}
+<h3>{% trans %}What?{% endtrans %}</h3>
+<p>{% trans site_name=site_name %}A third party ({{ origin }})
+has informed us that a {{ site_name }} API token associated
+with your account "{{ username }}" was found to be accessible at the following
+public url. We have revoked this token.{% endtrans %}</p>
+
+<p><a href="{{ public_url }}">{{ public_url }}</a></p>
+
+<h3>{% trans %}What should I do?{% endtrans %}</h3>
+<p>
+  {% trans %}First, we recommend that you inspect the public page where the token
+  appeared. Please investigate whether this is the result of an error on your side or if
+  someone else acquired and published your token.{% endtrans %}
+</p>
+<p>
+  {% trans site_name=site_name %}
+  If the token was not published by you, it means it was stolen, either from
+  you or from {{ site_name }}. We urge you to take all the necessary steps to secure all your
+  credentials.{% endtrans %}
+  {% trans site_name=site_name, href="https://python-security.readthedocs.io/packages.html#vulnerabilites-in-the-package-index" %}
+  In the case of a breach of the {{ site_name }} database, you will find informations
+  and support in the <a href="{{ href }}">python-security</a> documentation.{% endtrans %}
```

Comment:
If PyPI got breached and one of these tokens showed up on GitHub as a result, this may be the first indication we get. I'm not sure we'd already have information at that URL. I think the following are mutually exclusive:

* PyPI got breached, we found out and reset all tokens before they could make it on GitHub, and added some documentation to that effect.
* PyPI got breached, we didn't find out before they could make it on GitHub, an so there's no documentation to that effect.

In general, I don't think tokens leaking to GitHub would likely be indicative of a breach, so I'm not sure we need to devote this much discussion to it in the email.

Reply:
I’m ok with that, just thought it was a possibility.

I agree that the page wouldn’t exist yet when the mail is sent if we learn about the breach, but by the time people read their emails and click, a few hours/days may have passed, so the page will likely exist.

That being said, again, no problem dropping that